### ISSUE TITLE
Chucker retrun error

### ISSUE THREAD

**Issue by Zokirjon-Ashiraliyev** [syedmurtaza108/chucker-flutter#89]
![image](https://github.com/user-attachments/assets/6c70246d-6463-44fc-9585-b7aa40980caf)
This error available in chucker last and previouse version. When i remove chucker from interceptor, my project worked. How can i fix it. I do not have ApiResponse class in my project

**Comment by syedmurtaza108** [comment]
@Zokirjon-Ashiraliyev Seems for some reason your response couldn't be converted to JSON. Can you please share your response if possible?

**Comment by leonhomedigy** [comment]
I'm also seeing the same error after upgrading from 1.7.2 to 1.8.0.  I'll take a closer look tomorrow but the response object does implement toJson() and fromJson.

**Comment by syedmurtaza108** [comment]
@leonhomedigy @Zokirjon-Ashiraliyev To resolve it temporarily, I suggest to downgrade chucker_flutter to `1.7.2`. I will look into it further when I have some time.

To downgrade, use `chucker_flutter: 1.7.2` in pubspec.yaml

**Comment by 1mavic** [comment]
Hi!
I've update flutter to latest stable version 2.24.3. Having same issue&
Looks like trouble is in dio_intercepror.dart
in functions _saveResponse and _saveError when casting
 headers: response.requestOptions.headers.cast<String, String>()
 
 there can be null.
 When i change for  headers: {} no exceptions caught.

**Comment by 1mavic** [comment]
after additional investigation, found that trouble was if in header 'content-type':null

**Comment by syedmurtaza108** [comment]
@Zokirjon-Ashiraliyev @1mavic @leonhomedigy I attempted a fix in the branch `hotfix/json-encoding`.

Can any of you please add chucker_flutter in your pubspec through github like this, if someone verifies it I will publish it ASAP:

```
chucker_flutter:
    git:
      url: https://github.com/syedmurtaza108/chucker-flutter.git
      ref: hotfix/json-encoding
```

**Comment by syedmurtaza108** [comment]
Fixed deployed and is available at 1.8.1. Thank you all for reporting the issue.
